src/jit/lclvars.cpp

```diff
@@ -236,56 +236,13 @@ void Compiler::lvaInitTypeRef()
     lvaInitArgs(&varDscInfo);
 
 #if FEATURE_FASTTAILCALL
-
-    //-------------------------------------------------------------------------
-    // Calculate the argument register usage.
-    //
-    // This will later be used for fastTailCall determination
-    //-------------------------------------------------------------------------
-
-    unsigned argRegCount      = 0;
-    unsigned floatingRegCount = 0;
-    size_t   stackSize        = 0;
-
-    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc) {
-        if (varDsc->lvIsHfa())
-        {
-            floatingRegCount += varDsc->lvHfaSlots();
-        }
-        else
-        {
-            varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
-        }
-    };
-
-    unsigned   argNum;
-    LclVarDsc* curDsc;
-
-    for (curDsc = lvaTable, argNum = 0; argNum < varDscInfo.varNum; argNum++, curDsc++)
-    {
-        if (curDsc->lvIsRegArg)
-        {
-            incrementRegCount(curDsc);
-#if FEATURE_MULTIREG_ARGS
-            if (curDsc->lvOtherArgReg != REG_NA)
-            {
-                incrementRegCount(curDsc);
-            }
-#endif // FEATURE_MULTIREG_ARGS
-        }
-        else
-        {
-            stackSize += curDsc->lvArgStackSize();
-        }
-    }
-
     //-------------------------------------------------------------------------
     // Save the register usage information and stack size.
     //-------------------------------------------------------------------------
 
-    info.compArgRegCount      = argRegCount;
-    info.compFloatArgRegCount = floatingRegCount;
-    info.compArgStackSize     = stackSize;
+    info.compArgRegCount      = varDscInfo.intRegArgNum;
```

Comment:
I wonder if we should just replace `info.compArgRegCount` and `info.compFloatArgRegCount` with `codeGen->intRegState.rsCalleeRegArgCount` and `codeGen->floatRegState.rsCalleeRegArgCount`, which already exist.

Reply:
I agree with you.